### DISCUSSION TITLE
Async version of Langfuse.get_prompt

### DISCUSSION THREAD

**Discussion by antoniomdk** [langfuse/langfuse#3429] [Support] [answered]
### Describe the feature or potential improvement

The `get_prompt` function is mostly I/O bound, so It'd be great to have a version that doesn't block the event loop on stale/empty cache. An async lock would probably also be required to avoid race conditions when downloading and writing to the cache.

**Comment by marcklingen** [comment, marked as answer]
Thanks for sharing this.

I'd recommend to pre-fetch prompts when starting your application, thereby you do not run into the situation of an empty cache. Have a look here for an implementation example: https://langfuse.com/docs/prompts/get-started#optional-pre-fetch-prompts-on-application-start

**Reply by antoniomdk** [reply]
But, the cache has a TTL, right? I was under the impression that after the TTL, the client would be forced to re-download the prompt and block the event loop when calling `get_prompt`.

**Reply by marcklingen** [reply]
This was the case until recently. Now the stale cache is always returned instantly and then refetched in a background thread. I'd recommend to make sure that you have a single langfuse sdk client for your application instance to make sure that the cache works reliably.

changelog post: https://langfuse.com/changelog/2024-09-04-prompt-management-zero-latency

**Reply by antoniomdk** [reply]
Oh I see, that's wonderful!! If that's the case, there's no need for an async API. Thanks for the prompt reply!

**Reply by marcklingen** [reply]
Sure, I'm happy to help. We realized this week that this change eliminated the need for a separate async API, and we are very happy about it as well. :)

**Reply by marcellewi** [reply]
Hey! Was having same issue and came across this thread. Is there any async option nowadays?
